source/common/network/proxy_protocol.cc

```diff
@@ -143,11 +139,11 @@ bool ProxyProtocol::ActiveConnection::readLine(int fd, std::string& s) {
       }
     }
 
+    // Read the data upto and including the line feed, if available, but not past it.
+    // This should never fail, as search_index_ - buf_off_ <= nread, so we're asking
+    // only for bytes we have already seen.
     nread = recv(fd, buf_ + buf_off_, search_index_ - buf_off_, 0);
-
-    if (nread < 1) {
-      throw EnvoyException("failed to read proxy protocol");
-    }
+    ASSERT(size_t(nread) == search_index_ - buf_off_);
```

Comment:
This makes me a little nervous especially since ASSERT is elided in opt builds. If the \n was never found in a malformed message of MAX_PROXY_PROTO_LEN won't the loop end with buf_off_ == search_index_?  I guess we will still throw EnvoyException below in this case.